andela/travel_tool_back
# #160710678 Create travel team members on travela

#### What does this PR do?
Creates the endpoint for a travel admin to add travel team members

#### Description of Task to be completed?
- Create userRole join table for many-to-many relationship between user and role
- Modify role endpoints to adopt many-to-many relationship between user and role
- Create endpoint for posting travel team members
- Create middleware for validations
- Modify existing tests and add more tests for new code

#### How should this be manually tested?
- Clone the repository
- Enter into this branch - ft-create-travel-team-members-160710678
- Run 'yarn db:rollmigrate' to make migrations
- Start the application using 'make start'
- Using postman, access the endpoint 'PUT /api/v1/user/role/update' with the request body having 'center', 'email' and roleName(travel team member)
- You should see a success message that the request was successful


#### Any background context you want to provide?
To add a travel team member, ensure you have signed in into the system and your role is 10948 - super admin or 29187 - travel admin

#### What are the relevant pivotal tracker stories?
[160710678](https://www.pivotaltracker.com/story/show/160710678)

#### Screenshots (if appropriate)
<img width="1440" alt="screenshot 2018-10-16 at 5 40 14 pm" src="https://user-images.githubusercontent.com/25195857/47075054-97de8180-d1f3-11e8-91af-bd45c41d985f.png">


#### Questions:
N/A
